src/main/java/org/zalando/nakadi/plugin/auth/DefaultAuthorizationService.java

```diff
@@ -23,7 +25,8 @@ public boolean isAuthorizationAttributeValid(final AuthorizationAttribute author
 
     @Override
     public Subject getSubject() {
-        return () -> SecurityContextHolder.getContext().getAuthentication().getName();
+        return () -> Optional.ofNullable(
```

Comment:
Is it going to return `null` in case the request is not authenticated?

Reply:
Yes. Do you think it should be different? (The correct user will be covered by the plugin for OAuth2 requests) but for local nakadi, the user would be `null` (as it is presently))

Reply:
There are potentially other endpoints that are not authenticated, right? I guess "metrics" or "health" as well. We try to avoid returning `null` in this project. I see that changing it to `Option` could lead to extra effort and releasing a new `api` version (which takes hours). Let me know how you want to proceed. I would say we better change the API and return `Optional` or we document it somehow with `@Nullable`.